pyubee/__init__.py

```diff
@@ -314,36 +364,43 @@ def _base_url(self):
-        # pylint: disable=no-self-use
-        _LOGGER.debug('HTTP GET: %s', url)
-        req_headers = {'Host': self.host}
+        if hasattr(self, 'authenticator') and isinstance(self.authenticator, DigestAuthAuthenticator):
+            # We are using digest auth:
+            response = requests.get(url, timeout=HTTP_REQUEST_TIMEOUT, auth=HTTPDigestAuth(self.username, self.password))
+            return response
+        else:
+            # Use the rudimentary auth
+            """Do a HTTP GET."""
+            # pylint: disable=no-self-use
+            _LOGGER.debug('HTTP GET: %s', url)
+            req_headers = {'Host': self.host}
+
+            # Add custom headers.
+            for key, value in headers.items():
+                key_title = key.title()
+                req_headers[key_title] = value
+
+            # Add headers from authenticator.
+            for key, value in self._authenticator_headers.items():
+                key_title = key.title()
+                req_headers[key_title] = value
+
+            _LOGGER_TRAFFIC.debug('Sending request:')
+            _LOGGER_TRAFFIC.debug('  HTTP GET %s', url)
+            for key, value in req_headers.items():
+                _LOGGER_TRAFFIC.debug('  Header: %s: %s', key, value)
+
+            response = requests.get(url, timeout=HTTP_REQUEST_TIMEOUT, headers=req_headers)
+            _LOGGER.debug('Response status code: %s', response.status_code)
+
+            _LOGGER_TRAFFIC.debug('Received response:')
+            _LOGGER_TRAFFIC.debug('  Status: %s, Reason: %s', response.status_code, response.reason)
+            for key, value in response.headers.items():
+                _LOGGER_TRAFFIC.debug('  Header: %s: %s', key, value)
+            _LOGGER_TRAFFIC.debug('  Data: %s', repr(response.text))
+
+            return response
 
-        # Add custom headers.
-        for key, value in headers.items():
-            key_title = key.title()
-            req_headers[key_title] = value
-
-        # Add headers from authenticator.
-        for key, value in self._authenticator_headers.items():
-            key_title = key.title()
-            req_headers[key_title] = value
-
-        _LOGGER_TRAFFIC.debug('Sending request:')
-        _LOGGER_TRAFFIC.debug('  HTTP GET %s', url)
-        for key, value in req_headers.items():
-            _LOGGER_TRAFFIC.debug('  Header: %s: %s', key, value)
-
-        response = requests.get(url, timeout=HTTP_REQUEST_TIMEOUT, headers=req_headers)
-        _LOGGER.debug('Response status code: %s', response.status_code)
-
-        _LOGGER_TRAFFIC.debug('Received response:')
-        _LOGGER_TRAFFIC.debug('  Status: %s, Reason: %s', response.status_code, response.reason)
-        for key, value in response.headers.items():
-            _LOGGER_TRAFFIC.debug('  Header: %s: %s', key, value)
-        _LOGGER_TRAFFIC.debug('  Data: %s', repr(response.text))
-
-        return response
 
     def _post(self, url, data, **headers):
```

Comment:
Won't this need the alternate flow for the new Authenticator?

Reply:
Afaik this is handled by the one-liner `requests.get(...)`. I'll leave this open though, not 100% sure if some manual work is needed here.